torch/csrc/autograd/VariableTypeManual.cpp

```diff
@@ -194,6 +195,52 @@ void retain_grad(Tensor & self) {
   impl::get_autograd_meta(self)->retains_grad_ = true;
 }
 
+// Taken from codegened version
+Tensor fw_primal(const Tensor & self, int64_t level) {
+  auto& self_ = unpack(self, "self", 0);
+  std::shared_ptr<FwPrimalBackward> grad_fn;
+  if (compute_requires_grad( self )) {
+    grad_fn = std::shared_ptr<FwPrimalBackward>(new FwPrimalBackward(), deleteNode);
+    grad_fn->set_next_edges(collect_next_edges( self ));
+  }
+  #ifndef NDEBUG
+  c10::optional<Storage> self__storage_saved =
+    self_.has_storage() ? c10::optional<Storage>(self_.storage()) : c10::nullopt;
+  c10::intrusive_ptr<TensorImpl> self__impl_saved;
+  if (self_.defined()) self__impl_saved = self_.getIntrusivePtr();
+  #endif
+  auto tmp = ([&]() {
+    at::AutoNonVariableTypeMode non_var_type_mode(true);
+    // Modified from original codegen
+    return self_.view(self_.sizes());
+    // End modified from original codegen
+  })();
+  c10::optional<std::function<at::Tensor(const at::Tensor&)>> func=c10::nullopt;
+  if (false || !self.unsafeGetTensorImpl()->support_as_strided()) {
```

Comment:
`false ||`? leftover debug code?

Reply:
This comes from the codegen. I left it as is to make future updates easy.